week4/assignment/week4_assignment/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
+import NamePage from './SignUp/NamePage';
+import PasswordPage from './SignUp/PasswordPage';
+import HobbyPage from './SignUp/HobbyPage';
+import { useNavigate } from 'react-router-dom';
+import { instance } from '../api/axios';
+import { Theme } from '../styles/Themes';
+import styled from '@emotion/styled';
+
+interface FormData {
+  id: string;
+  password: string;
+  hobby: string;
+}
+
+type ChangeType = (field: keyof FormData, value: string) => void;
+
+const SignUp = () => {
+  const [step, setStep] = useState(1);
+  const [formData, setFormData] = useState<FormData>({
+    id: '',
+    password: '',
+    hobby: '',
+  });
+  const navigate = useNavigate();
+
+  const ChangeField: ChangeType = (field, value) => {
+    setFormData((prev) => ({ ...prev, [field]: value }));
+  };
+
+  const ChangeNext = () => {
+    if (step < 3) setStep((prev) => prev + 1);
+  };
+
+  //회원가입 API 요청
+  const ApiSignup = async () => {
+    try {
+      const response = await instance.post('/user', {
+        username: formData.id,
+        password: formData.password,
+        hobby: formData.hobby,
+      });
```

Comment:
```typescript
interface FormData {
  username: string; // 기존 id -> username으로 수정
  password: string;
  hobby: string;
}
```

와 같이 변수 명을 API 요청 변수명과 맞춘다면


```suggestion
      const response = await instance.post('/user', { ...formData });
```

과 같이 훨씬 간결하고 깔끔하게 작성할 수 있을 것 같습니다!